File: Python/Task345.py
#оголошуємо масив (список) A та заповнюємо
#його введеними користувачем числами
A = []
print('Введіть розмір масиву A')
size_a = int(input())
print('Введіть елементи масиву A')
for i in range(0,size_a):
    A.append(float(input()))
#оголошуємо масив (список) B та заповнюємо
#його введеними користувачем числами
B = []
print('Введіть розмір масиву B')
size_b = int(input())
print('Введіть елементи масиву B')
for i in range(0,size_b):
    B.append(float(input()))
#оголошуємо результуючий масив (список) 
#та організовуємо його заповнення
C = []
C.append(A[0])
for i in range(1,size_a):
    present = False
    for el in C:
        if el == A[i]:
            present = True
            break    
    if not present: 
        C.append(A[i])
for i in range(0,size_b):
    present = False
    for el in C:
        if el == B[i]:
            present = True
            break    
    if not present : 
        C.append(B[i])
#Виводимо масиви
print('Масив A')
for el in A:
    print(" %8i" % el, end=' ')
print() #Перехід на новий рядок
print('Масив B')
for el in B:
    print(" %8i" % el, end=' ')
print() #Перехід на новий рядок
print('Масив C')
for el in C:
    print(" %8i" % el, end=' ')
print() #Перехід на новий рядок

File: Pascal/Task345.pas
program Task345;
var sizeA, sizeB, sizeC, i, j: integer;
    A, B : array [1..1000] of integer;
    C : array [1..2000] of integer;
    present : integer;
begin
    writeln('Введіть розмір масиву A');
    readln(sizeA);
    writeln('Введіть елементи масиву A');
    for i := 1 to sizeA do readln(A[i]);
    writeln('Введіть розмір масиву B');
    readln(sizeB);
    writeln('Введіть елементи масиву B');
    for i := 1 to sizeB do readln(B[i]);
    sizeC := 0;{в sizeC будемо підраховувати кількість 
        елементів доданих в масив C   }
    C[1] := A[1]; 
    sizeC := sizeC + 1; {збільшуємо значення лічильника}
    for i := 1 to sizeA do
    begin
        present := 0;{ встановлюємо значення 0 для
        чергового елемента, якщо після циклу це значення
        залишиться рівним 0, це буде означати, що такого
        елемента в результуючому масиві ще немає}
        for j := 1 to sizeC do
            if A[i]=C[j] then
            begin
                present := 1;
                break
            end;
        {внутрішній цикл по j містить в тілі лише один 
        оператор розгалуження, begin-end не використовуємо,
        тут закінчується  тіло цього циклу}
        if present = 0 then
        begin
            sizeC := sizeC + 1;
            C[sizeC] := A[i]
        end
    end;
    {Переревіряємо котрі з елементів масиву B
    потрібно записати в масив C}
    for i := 1 to sizeB do
    begin
        present := 0;
        for j := 1 to sizeC do
            if B[i]=C[j] then
            begin
                present := 1;
                break
            end; {тут закінчується тіло циклу по j}
        if present = 0 then
        begin
            sizeC := sizeC + 1;
            C[sizeC] := B[i]
        end
    end;
    writeln('Масив A:');
    for i:=1 to sizeA do write(A[i]:8);
    writeln();
    writeln('Масив B:');
    for i:=1 to sizeB do write(B[i]:8);
    writeln();
    writeln('Масив C:');
    for i:=1 to sizeC do write(C[i]:8);
    writeln();
end.
